### PR TITLE
Changed README.md and package.json to get a working dev and production environment and added data-cy properties to some components for manipulation in Cypress tests

### PR DESCRIPTION
**Pull request by NikAiyer**
Added yarn db:migrate as a command in the root package.json to make it more convenient to run. 

Added yarn db:migrate and yarn build:relay under the Development tab in the README, as these are necessary to get yarn dev running.

Added a data-cy property 'start-demo-button' to BeginDemoModal, passing it through props as buttonName.

Added a data-cy property "sidebar-header" to NewMeetingSideBar

### CODE REVIEW

**Comment by mattkrick** [comment]
LGTM, i noticed  you closed #3612  and #3615, is that in favor of an upcoming PR?

**Comment by NikAiyer** [comment]
Yeah I thought it might be better to make smaller PRs. I also made some unnecessary changes in those old PRs that I didn't want to put in the request.

File: packages/client/components/BaseButton.tsx [hunk 1]
```diff
@@ -41,6 +41,7 @@ export interface BaseButtonProps extends PlainButtonProps {
   size?: 'small' | 'medium' | 'large'
   children?: ReactNode
   className?: string
+  buttonName?: string
```
Comment (mattkrick) [inline packages/client/components/BaseButton.tsx]: nit-- it'd be nice to have a prop name used for naming testing attributes for all components, not just buttons. maybe we just have a `cy` or `dataCy` prop? that way it'll be the same across divs, forms, buttons, etc.

File: packages/client/components/PrimaryButton.tsx [hunk 2]
```diff
@@ -6,11 +6,12 @@ import {Radius} from 'types/constEnums'
 import {Elevation} from 'styles/elevation'
 
 const StyledBaseButton = styled(BaseButton)((props: BaseButtonProps) => {
-  const {disabled, waiting} = props
+  const {disabled, waiting, buttonName} = props
   const visuallyDisabled = disabled || waiting
   return {
     backgroundImage: visuallyDisabled ? PALETTE.GRADIENT_WARM_LIGHTENED : PALETTE.GRADIENT_WARM,
     borderRadius: Radius.BUTTON_PILL,
+    buttonName,
```
Comment (mattkrick) [inline packages/client/components/PrimaryButton.tsx]: can remove, these are just for styles. emotion will pass all props through to the wrapped element

Comment (NikAiyer) [reply packages/client/components/PrimaryButton.tsx]: Ok I'll remove that.

File: packages/client/components/BaseButton.tsx (hunk 1 above)
Comment (NikAiyer) [reply packages/client/components/BaseButton.tsx]: Yeah that makes sense to me. I'll make that change now.

**Pull request merged**